k6-runner/Dockerfile

```diff
@@ -1,12 +1,13 @@
-FROM --platform=${BUILDPLATFORM} golang:1.23-alpine3.19 as builder
+FROM --platform=${BUILDPLATFORM} golang:1.24-alpine3.22 AS builder
```

Comment:
changes in this file are unrelated to the e2e image builder, but this is triggered too because the `Makefile` has changed and it fails with what is currently in `main` branch because the xk6@latest requires go 1.24

```
#10 0.469 go: downloading go.k6.io/xk6 v1.1.4
#10 0.533 go: go.k6.io/xk6/cmd/xk6@latest: go.k6.io/xk6@v1.1.4 requires go >= 1.24.0 (running go 1.23.4; GOTOOLCHAIN=local)
#10 ERROR: process "/bin/sh -c go install go.k6.io/xk6/cmd/xk6@latest" did not complete successfully: exit code: 1
```